utils/fw_tunnel_updater.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import getpass
+import sys
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+from rffe_nuttx_lib import RFFEFWUpdate
+from rffe_tunnel_lib import SSHTunnel
+
+
+def load_rack_config(file_path):
+    with open(file_path) as f:
+        return json.load(f)
+
+def get_getway(rack_id):
+    if rack_id == "TL":
+        return "ia-20rabpmtl-co-iocsrv.lnls-sirius.com.br"
+    else:
+        return f"ia-{rack_id}rabpm-co-iocsrv.lnls-sirius.com.br"
+
+def tunnel_forward_fw_update(gateway, ip, password, fw_file, version, key_path):
+    tunnel = SSHTunnel(
+            host=gateway,
+            username="lnls-bpm", 
+            key_path=key_path,
+            remote_host=ip,
+            remote_port=9090,
+            key_pass=password
+            )
+
+    try:
+        ssh_channel = tunnel.connect()
+        rffe = RFFEFWUpdate(ip_addr='localhost', sock=ssh_channel)  
+        
+        print(f"[INFO] Connected on ip {ip}. Flashing firmware...")
+        rffe.reprogram(fw_file, version)
+        print(f"[INFO] Firmware update completed on ip {ip}")
+        rffe.close()
+    except Exception as e:
+        print(f"[ERROR] Firmware update failed on {ip}: {e}")
+    finally:
+        tunnel.close()            
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="RFFE Firmware Update Tool")
+    parser.add_argument("--json-in", type=str, help="Path to rack configuration JSON file", required=True)
+    parser.add_argument("--v", type=str, help="Firmware version", required=True)
+    parser.add_argument("--bin-file", type=str, help="Path to .bin firmware file", required=True)
+    parser.add_argument('-l','--list', nargs='+', help="Rack IDs to update", required=False)
+    parser.add_argument("--key-path", type=str, help="Path to SSH private key", required=False)
+    args = parser.parse_args()
+        
+    racks = load_rack_config(args.json_in)
+    
+    ssh_password = getpass.getpass("Enter SSH private key password: ")
+
+    for rack_id in args.list:
+        rack = next((rack for rack in racks if str(rack["rack"]) == rack_id), None)        
+        if not rack:
+            print(f"[ERROR] Rack {rack_id} not found in {args.json_in}")
+            continue
+
+        gateway = get_getway(rack_id)
+        max_threads = 11
```

Comment:
I think it would be better to add an argument to set the maximum threads instead of hardcoding it here.